src/main/java/edu/ucla/library/prl/harvester/handlers/UpdateJobHandler.java

```diff
@@ -130,6 +136,19 @@ Future<UpdateResponse> updateSolr(final Tuple3<Job, Job, Institution> aData) {
         });
     }
 
+    /**
+     * @param anOldJob A Job representing an original state
+     * @param aNewJob A Job representing a new, updated state
+     * @return True if new job has sets not contained in old job
+     */
+    private boolean hasNewSets(final Job anOldJob, final Job aNewJob) {
+        return getDifference(aNewJob.getSets(), anOldJob.getSets()).isEmpty() ||
+                !anOldJob.getSets().isEmpty() && aNewJob.getSets().isEmpty() &&
+                        (getDifference(OaipmhUtils.listSets(myVertx, anOldJob.getRepositoryBaseURL(),
+                                myOaipmhClientHttpTimeout, myHarvesterUserAgent).map(OaipmhUtils::getSetSpecs).result(),
+                                anOldJob.getSets()).isEmpty());
```

Comment:
Per the Javadoc for Future#result:
> The result of the operation. This will be null if the operation failed.

That is not acceptable in this instance, since in the event that OaipmhUtils#listSets returns a FailedFuture, getDifference will be called with a `null` anOldList, throwing an NPE. That must be avoided.